### DISCUSSION TITLE
Add physbone and colliders to existing avatar

### DISCUSSION THREAD

**Discussion by npprs** [bdunderscore/modular-avatar#1181] [Q&A]
I am looking to add physbone configuration as well as corresponding collider to the avatar via MA. I haven't been able to achieve this via Merge Armature nor Replace Object. I got close to achieving what I need but it came with too many caveats and feels really hacky.

Base Armature
```
Hips
  Leg_L
    Knee_L
      Toe_L
        Toe_L.001
           ...
  Leg_R
    Knee_R
      Toe_R
        Toe_R.001
           ...
```

Objective A: PhysBone and Collider needs to be setup for the base avatar on build.

Observation: Any PhysBone inside `Merge Armature` will not get merged. Only `Replace Object` will work. In addition, PhysBones and Collider will need to target `none`. If it targets the bones `Merge Armature` will not merge properly. PhysBone and Collider preview will be incorrect in edit mode until build since it has to target `none`.
```
Hips
  Leg_L
    Knee_L
      Toe_L (Collider)
        Toe_L.001 (PhysBone w/ Toe_R Collider)
           ...
  Leg_R
    Knee_R
      Toe_R (Collider)
        Toe_R.001 (PhysBone w/ Toe_L Collider)
           ...
```

Objective B: User should be able to preview, create, update animations with MA.

Observation: This can be done if the prefab has `Animator`, `MA Merge Animator`, and `Armature w/ Merge Armature`. If no armature is provided all animation will be marked as missing.

Objective C: User should be able to enable and disable PhysBones and Colliders.

Observation: Enable and Disable PhysBone and Collider animation references are broken because there is nothing to reference inside `Merge Armature`. Referencing `Replace Object` will simply not work. I manually edit the animations paths to enable and disable PhysBones and colliders. Unfortunately the mesh stretches for an instant before snapping back to normal whenever toggle is triggered.

Here is an example structure of my test setup:
```
Avatar
  Armature
  Body
  Foot Control (Animator, MA Merge Animator, MA Paramater, MA Menu Installer, MA Menu Group)
    Armature
      Hip (Merge Armature)
        ...
    MA Replace
      Toe_L.001
      Toe_R.001
      Toe_L
      Toe_R
```

**Comment by bdunderscore** [comment]
This is an open feature request: https://github.com/bdunderscore/modular-avatar/issues/812
I have some ideas, but mainly need to decide whether to implement a general "move this component somewhere else" thing, or a platform-independent way of declaring dynamics.
